graph/patterns/operations.md

```diff
@@ -2,26 +2,34 @@
 
 ## Solution
 
-To address these use cases, API designers might use operational resources such as functions or actions.
-If the operation doesn't have any side effects and MUST return a single instance of a type or a collection of instances, then the designer SHOULD use the OData function; otherwise, the designer can model the operation as an action.
+To address these use cases, API designers can use operational resources such as functions or actions.
+If the operation doesn't have any side effects and MUST return a single instance of a type or a collection of instances, then the designer MUST use OData functions; otherwise, the designer can model the operation as an action.
 
 ## When to use this pattern
 
 The operations pattern is well suited to use cases that cannot be modeled as a single HTTP method on a resource and require either multiple round trips to complete a single logical operation or produce one or multiple side effects.
 
+The operation pattern might be justified when a modeling operation represents one or combination of the following:
+
+- a change of a resource (i.e., increment the value of a property) rather than a state (i.e., the final value of the property)
+- complex processing logic that shouldn't be exposed to the client
+- operation parameters might convey a restricted set of option (i.e., a report that can has to specify a date range)
+- the operation leverage some service-side data not exposed to (or easily retrieved in context by) the user.
+
 You can consider related patterns such as [long running operations](./long-running-operations.md) and [change tracking](./change-tracking.md).
 
 ## Issues and considerations
 
-- Microsoft Graph does NOT support unbound actions or functions. Bound actions and functions are invoked on resources matching the type of the binding parameter. The binding parameter can be of any type, and it MAY be Nullable. For Microsoft Graph, actions and functions must have the `isBound="true"` attribute. The first parameter is the binding parameter.
+- Microsoft Graph does NOT support unbound actions or functions. Bound actions and functions are invoked on resources matching the type of the binding parameter. The binding parameter can be of any type, and parameter value MAY be Nullable. 
+For Microsoft Graph, actions and functions must have the `isBound="true"` attribute. The first parameter is the binding parameter.
 
 - Both actions and functions support overloading, meaning a schema might contain multiple actions or functions with the same name. The overload rules as per the OData [standard](http://docs.oasis-open.org/odata/odata-csdl-xml/v4.01/odata-csdl-xml-v4.01.html#sec_FunctionOverloads) apply when adding parameters to actions and functions.
```

Comment:
I think that's true for REST.  But not sure that overloading works on all of our SDK platforms.  Please check with @baywet

Reply:
now that we switched to using a parameter set class for actions/functions in Java (almost 2 years ago now), I believe all of our SDKs support overloading. And all Kiota based SDKs have been designed with that requirement in mind. Thanks for double checking.

Reply:
That is awesome! Thanks for the update.